src/main/java/frc/robot/subsystems/DrivetrainSubsystem.java

```diff
@@ -260,6 +276,20 @@ public void periodic() {
                         chassisSpeeds.vyMetersPerSecond = yPID.calculate(getPose().getTranslation().minus(translationTarget).getY());
                 }
 
+                if(balanceTarget != null ) {
+                        if (Math.abs(getGyroscopePitch().getDegrees()) >= AUTO_BALANCE_TOLERANCE_DEGREES) {
+                                chassisSpeeds.vyMetersPerSecond = yPID.calculate(getGyroscopePitch().minus(balanceTarget).getRadians());
+                        } else if (Math.abs(getGyroscopeRoll().getDegrees()) >= AUTO_BALANCE_TOLERANCE_DEGREES) {
+                                chassisSpeeds.vxMetersPerSecond = xPID.calculate(getGyroscopeRoll().minus(balanceTarget).getRadians());
+                        }
```

Comment:
It occurred to me last night that for each of x and y we need to set the speeds to 0 when in the tolerance otherwise they will remember their last value before within tolerance